app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,74 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+
+  # GET /categories or /categories.json
+  def index
+    @categories = Category.all
+  end
+
+  # GET /categories/1 or /categories/1.json
+  def show
+    @category = Category.find(params[:id])
+    @budgets = @category.budgets
+    @total_amount = @budgets.sum(&:amount)
+  end
+
+  # GET /categories/new
+  def new
+    @category = Category.new
+  end
+
+  # GET /categories/1/edit
+  def edit; end
+
+  # POST /categories or /categories.json
+  def create
+    @category = Category.new(category_params)
+    @catagory.author = current_user
```

Comment:
- [ ] [REQUIRED] When I tried to add a new category I got a `NoMethodError`.

### Here is the error image
<img width="1432" alt="Screenshot 2023-03-31 at 18 03 14" src="https://user-images.githubusercontent.com/58692590/229195237-bc90526a-7af3-4a72-a4b3-30de39bf0f9d.png">

### This is why I think we are getting the error 
The NoMethodError you're seeing is because the @category object is nil, which means that it doesn't exist or hasn't been initialized. When you try to call the author= method on @category, you're essentially trying to call a method on a nil object, which results in the NoMethodError.

### Here are some possible reasons why @category might be nil:

The @category variable hasn't been initialized: Make sure that you've initialized the @category variable before trying to use it. In Rails, this is typically done in the controller action or in a before_action method.

The @category variable was not passed to the view: If you're trying to use the @category variable in a view template, make sure that it has been passed to the view from the controller action. You can do this by setting the @category variable in the controller action and then passing it to the view using an instance variable.

There is an error in your code that prevents @category from being initialized: Check your code for any errors that might be preventing @category from being initialized properly. For example, if you have a conditional statement that prevents @category from being initialized in certain cases, make sure that the condition is correct and that @category is initialized in all necessary cases.